synapse/util/stringutils.py

```diff
@@ -26,6 +27,15 @@
 # https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-register-email-requesttoken
 client_secret_regex = re.compile(r"^[0-9a-zA-Z\.\=\_\-]+$")
 
+# https://matrix.org/docs/spec/client_server/r0.6.1#matrix-content-mxc-uris,
+# together with https://github.com/matrix-org/matrix-doc/issues/2177 which basically
+# says "there is no grammar for media ids"
+#
+# The server_name part of this is purposely lax: use parse_and_validate_mxc for
+# additional validation.
+#
+mxc_re = re.compile("^mxc://([^/]+)/([^/#?]+)$")
```

Comment:
Should this be `MXC_RE` (as `VALID_HOST_REGEX` is).

Reply:
well if we're doing that, let's call it `MXC_REGEX` ?

anyway, sure.